crates/sage-cli/src/main.rs

```diff
@@ -148,31 +183,37 @@ impl Runner {
             .sn
             .then_some(self.parameters.quant.tmt_settings.level);
 
-        let spectra = chunk
+        let (valids, errors): (Vec<_>, Vec<_>) = chunk
             .par_iter()
-            .map(|path| sage_core::read_mzml(path, sn))
-            .collect::<Vec<_>>();
+            .map(|path| {
+                sage_core::read_mzml(path, sn)
+                    .with_context(|| format!("Failed to process `{path}`"))
+            })
+            .partition(Result::is_ok);
+
+        if !errors.is_empty() {
+            let errors: Vec<_> = errors
+                .into_iter()
+                .map(Result::unwrap_err)
+                .map(|err| RunnerError::FileProcessingError { source: err })
+                .collect();
+
+            anyhow::bail!(RunnerError::ChunkProcessingError {
+                chunk_idx,
+                source: errors
+            });
+        }
```

Comment:
Returns all the problematic files from the first problematic chunk. Not great as it may already have processed files from previous chunks, but still an improvement as it fails after failing to process a file.